pkg/plugins/ca/provided/ca_cert_validator.go

```diff
@@ -30,18 +30,18 @@ func validateCaCert(signingPair util_tls.KeyPair) (verr validators.ValidationErr
 		verr.AddViolation("cert", fmt.Sprintf("not a valid x509 certificate: %s", err))
 		return
 	}
-	//if cert.Issuer.String() != cert.Subject.String() {
-	//	verr.AddViolation("cert", "certificate must be self-signed (intermediate CAs are not allowed)") // Envoy constraint
-	//}
+	if cert.Issuer.String() != cert.Subject.String() {
+		verr.AddViolation("cert", "certificate must be self-signed (intermediate CAs are not allowed)") // Envoy constraint
+	}
```

Comment:
I don't think this should be a part of the PR. I will be happy to approve this one as a separate PR once we have a confirmation with E2E testing.

Reply:
Hi @jakubdyszkiewicz It is not part of PR , if you check the files changed. Also if you see above changes , it is shown as outdated